python/tests/plugins/test_gen_plant_uml.py

```diff
@@ -1,169 +1,267 @@
-                self._assert_sequence_diagram_content(puml_content)
+        with temporary_test_file(TEST_PUML_ARCH_YAML, suffix=YAML_FILE_EXTENSION) as plugin_yaml:
+            result = puml_sequence(plugin_yaml.name)
+            self.assertEqual(result.status_code, PluginExecutionStatusCode.SUCCESS)
+            self._check_sequence_diagram_usecase_one(result.output())
 
     def test_puml_sequence_diagram_to_file(self):
-        with TemporaryDirectory() as temp_directory:
-            with NamedTemporaryFile(dir=temp_directory, mode="w", suffix=".yaml") as plugin_yaml:
-                plugin_yaml.write(TEST_PUML_ARCH_YAML)
-                plugin_yaml.seek(0)
-
-                # Get the rng temp AaC file name, but with a puml extension
-                expected_puml_file_path = os.path.basename(plugin_yaml.name).replace(".yaml", f"_{SEQUENCE_STRING}.puml")
-
-                result = puml_sequence(plugin_yaml.name, temp_directory)
-                self.assertIn(expected_puml_file_path, "\n".join(result.messages))
-                self.assertEqual(result.status_code, PluginExecutionStatusCode.SUCCESS)
-
-                temp_directory_files = os.listdir(temp_directory)
-
-                self.assertIn(expected_puml_file_path, temp_directory_files)
-
-                with open(os.path.join(temp_directory, expected_puml_file_path)) as generated_puml_file:
-                    generated_puml_file_content = generated_puml_file.read()
-                    self._assert_diagram_contains_uml_boilerplate(generated_puml_file_content)
-                    self._assert_sequence_diagram_content(generated_puml_file_content)
-
-    def _assert_component_diagram_content(self, component_diagram_content_string: str):
+        with (TemporaryDirectory() as temp_directory,
+              temporary_test_file(TEST_PUML_ARCH_YAML, dir=temp_directory, suffix=YAML_FILE_EXTENSION) as plugin_yaml):
+            result = puml_sequence(plugin_yaml.name, temp_directory)
+
+            full_output_dir = os.path.join(temp_directory, SEQUENCE_STRING)
+            self.assertIn(full_output_dir, result.output())
+            self.assertEqual(result.status_code, PluginExecutionStatusCode.SUCCESS)
+
+            expected_puml_file_paths = [
+                _get_generated_file_name(plugin_yaml.name, SEQUENCE_STRING, name)
+                for name in [TEST_PUML_USE_CASE_ONE_TITLE, TEST_PUML_USE_CASE_TWO_TITLE]
+            ]
+            temp_directory_files = os.listdir(full_output_dir)
+            for path in expected_puml_file_paths:
+                basename = os.path.basename(path)
+                self.assertIn(basename, temp_directory_files)
+
+                _, *parts = os.path.splitext(basename)[0].split("_")
+                check_generated_file_contents(path, self.get_checker_from_filepath("_".join(parts[-2:]), SEQUENCE_STRING))
+
+    def get_checker_from_filepath(self, path: str, puml_type: str) -> Callable:
```

Comment:
This should be a private function
```suggestion
    def _get_checker_from_filepath(self, path: str, puml_type: str) -> Callable:
```

Reply:
Resolved in 5692ca9